src/java.desktop/share/classes/javax/swing/plaf/nimbus/skin.laf

```diff
@@ -268,7 +268,7 @@
       </uiComponent>
       <uiComponent opaque="false" type="javax.swing.JButton" name="Button" ui="ButtonUI" subregion="false">
          <stateTypes/>
-         <contentMargins top="6" bottom="6" left="14" right="14"/>
+         <contentMargins top="6" bottom="6" left="0" right="0"/>
```

Comment:
> I also tested for non-HTML text after the changes, and the changes do not affect normal text.

And this is true for this Nimbus case too ? Whereas your code update in BasicButtonUI is checking for HTML, I don't see how it could *not* change normal text in the Nimbus case.

Reply:
I thought the same thing. In the test, I forced the L&F to nimbus only and tested with the left and right insets set to 0 and 14. The text appears normally and identically for both cases  (HTML text and non-HTML text). I also tried changing the size of the button from 37x37 to 100x100 and 200x200. The appearance was the same for all cases.

However, I just tried using different text from the tests before and I do get cases where text gets snipped to "..." when the new insets wouldn't. So, you're right that normal text is affected. I just used text that wasn't long enough in testing.

The reason I went this route for the fix was because in NimbusStyle, the insets are retrieved but the property key but the button component for BasicHTML is not set yet and returns null. So, I can't use this method to set the insets to 0 since the property key has not been set in the stack yet.

I finally found the applicable class of where to make the edits yesterday. I just made the changes similar to the other L&Fs.